### PR TITLE
Remove SyncQueue and migrate more ES6 classes

### PR DESCRIPTION
**Pull request by willeastcott**
* `src/scene/sync-queue.js` was left over from a [long reverted PR](https://github.com/playcanvas/engine/pull/1437) so this PR deletes it. It's definitely not used anywhere in the codebase.
* Splits `SceneRegistryItem` into its own module.
* Updates a few more classes to ES6.

I confirm I have signed the [Contributor License Agreement](https://docs.google.com/a/playcanvas.com/forms/d/1Ih69zQfJG-QDLIEpHr6CsaAs6fPORNOVnMv5nuo0cjk/viewform).

### CODE REVIEW

**Comment by willeastcott** [comment]
Hey @vkalpias, I'm considering making another couple of changes:

1. Move `SortedLoopArray` into `src/core`.
2. Removing `SortedLoopArray` from the `pc` namespace, effectively making it private to the engine codebase? It's JSDocs have always been private.

Any objections?

File: src/index.js [hunk 1]
```diff
@@ -22,7 +22,6 @@ export { IndexedList } from './core/indexed-list.js';
 export { Tags } from './core/tags.js';
 export { Timer, now } from './core/time.js';
 export { URI, createURI } from './core/uri.js';
-export { RefCountedObject } from './core/ref-counted-object.js';
```
Comment (mvaligursky) [inline src/index.js:25]: so a class does not need to be added here if it's private?

Comment (willeastcott) [reply src/index.js:25]: Correct. The `index.js` module defines the classes/function/properties that are exported to the `pc` namespace.

**Review by mvaligursky — APPROVED** [review]

**Comment by vkalpias** [comment]
> Hey @vkalpias, I'm considering making another couple of changes:
> 
> 1. Move `SortedLoopArray` from `src/framework/utils` to `src/core`.
> 2. Removing `SortedLoopArray` from the `pc` namespace, effectively making it private to the engine codebase. Its JSDocs have always been private.
> 
> Any objections?

Sounds good

**Comment by willeastcott** [comment]
OK thanks, done!

**Pull request merged**